[FIX] sale: default on product change in configurator

nle-odoo opened the pull request:
When a product is set on a line and we change product with the product
configurator wizard, we will use the attributes values of the previous
product.

But then since the attributes are the ones of wrong product, not
attribute is selected and if eg. we have radio button with one option
and custom value, it will never be selectable (because the radio button
is hidden and unselected).

In this commit, we do not use the current variant values when we change
product in the product configurator wizard.

note: this commit also increases the limit of (custom values) inside the
product configurator wizard, for an arbitrary 10000 limit. Without this
change, each opening of the wizard would only keep 40 customs values and
discard the other ones.

opw-2333091

awa-odoo commented on addons/sale/views/variant_templates.xml:
```diff
@@ -58,7 +58,7 @@
                                             <input type="radio"
                                                 t-attf-class="js_variant_change radio_input #{variant_id.attribute_id.create_variant}"
-                                                t-att-checked="value_id in combination"
+                                                t-att-checked="value_id in combination or single_and_custom"
```
Hello.

I would have to dive back into that code to give you more insights but I'm unsure how this fixes an issue, even after reading your commit message. Could you elaborate?
Also, does this only apply to the "radio" type? The "select" and "color" types have similar code.

nle-odoo replied:
you are right it would also apply for `color` and maybe `select` since the issue is that nothing is selected and the field is not shown because of:

```xml
t-attf-class="list-inline  #{'d-none' if single_and_custom else ''}">
```

I expanded on the issue on a direct message.

nle-odoo commented:
Hi @awa-odoo 


It seems that with a radio with only one option, once a product is set on a line, when we open the product configurator we will have in context the original :

https://github.com/odoo/odoo/blob/8c3ac0a4b52b1d7fec3d062cce2f81626f9e7861/addons/sale_product_configurator/static/src/js/product_configurator_controller.js#L113-L127

and because of that, we do not do `product._get_first_possible_combination()` of:

https://github.com/odoo/odoo/blob/8c3ac0a4b52b1d7fec3d062cce2f81626f9e7861/addons/sale_product_configurator/views/templates.xml#L13

and the options do not get value, so for the case of "one radio button" that is not displayed, it is not possible to select it since they are not shown.

another thing I tried was to detect if we changed the product, I have updated the code of this PR for that since it seems more clean in my opinion.

awa-odoo commented:
Hello @nle-odoo 

Thanks for the full explanation, I get what is going on now.
It's a bit of a corner case that is indeed not well handled in the current code.

I think I like your latest fix proposal better.
I tested it a bit on a runbot and it seems to work fine for existing (more simple) use cases.

Thanks for looking into it.

nle-odoo reviewed (commented):
I changed the commit message a little.

This should be alright to merge now:

robodoo r+
